synapse/storage/data_stores/main/roommember.py

```diff
@@ -433,14 +396,40 @@ def get_rooms_for_user_with_stream_ordering(self, user_id):
             the rooms the user is in currently, along with the stream ordering
             of the most recent join for that user and room.
         """
-        rooms = yield self.get_rooms_for_user_where_membership_is(
-            user_id, membership_list=[Membership.JOIN]
-        )
-        return frozenset(
-            GetRoomsForUserWithStreamOrdering(r.room_id, r.stream_ordering)
-            for r in rooms
+        return self.db.runInteraction(
+            "get_rooms_for_user_with_stream_ordering",
+            self._get_rooms_for_user_with_stream_ordering_txn,
+            user_id,
         )
 
+    def _get_rooms_for_user_with_stream_ordering_txn(self, txn, user_id):
+        if self._current_state_events_membership_up_to_date:
+            sql = """
+                SELECT room_id, e.stream_ordering
+                FROM current_state_events AS c
```

Comment:
As per `get_rooms_for_user_with_stream_ordering` docstring, this can be called against remote users which are not in the new table. Can add a docstring here too.

Reply:
ok, makes more sense now, thanks.

I'd add some comments to the effect of:

* we can't use `local_current_membership` because we want it to work for remote users
* we only care about rooms that we are currently participating in, so `current_state_events` is good enough.